packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -290,7 +340,15 @@ class LinearProgressIndicator extends ProgressIndicator {
     this.minHeight,
     super.semanticsLabel,
     super.semanticsValue,
-    this.borderRadius = BorderRadius.zero,
+    this.borderRadius,
+    this.stopIndicatorColor,
+    this.stopIndicatorRadius,
+    this.trackGap,
+    @Deprecated(
+      'Use ProgressIndicatorTheme to customize the ProgressIndicator appearance. '
+      'This feature was deprecated after v3.26.0-0.1.pre.'
+    )
+    this.year2023 = false,
```

Comment:
Should here default to true if the flag called `year2023`? I think from our discussion in the design doc, we want the flag defaults to a value which doesn't cause breaking changes. Like in the doc, the original flag is called `gappedStyle` and we want it default to false to hide the latest look:).

Reply:
Ohhh i was thinking opposite so we users just can just turn off new design manually but setting to `true` by default makes sense if we don't wanna any breaking changes initially. 

I updated the PR and looks like all existing tests are green 🎉

Reply:
I'll write migration guide for the new flag as we're deprecating right away in this PR. WDYT?

Reply:
A migration guide would be super helpful! Thanks a lot! We can also create an example to toggle the two progress indicator themes so that when this flag is removed in the future, users can copy and paste the old theme from the example if they really want.

Reply:
As i started to write the migration guide i realized they've relatively small updates compare to Sliders. I think it'd be best to write single migration guide for both progress indicators (Linear and Circular).  Now that we've a clear direction with the flag. 

We can land this PR and I'm already working on circular progress indicator update PR. I'll include both in a single progress indicators migration guide.

Reply:
Sounds good! Since the flag will not be removed immediately, I think it's fine to have a single migration guide for both progress indicators! Thanks a lot!